### ISSUE TITLE
htmx won't insert <style> tag

### ISSUE THREAD

**Issue by jsonberry** [bigskysoftware/htmx#1771]
htmx version: 1.9.5

Here's a barebones example for when I attempt to get a `style` tag as the main node it does not appear in the DOM.

I've tried various techniques and nothing has worked so far. There is one workaround listed below.

I suspect `htmx` treats `style` special because I've noticed that it will insert a `style` tag into the `head` on load, maybe there's something funky going on there.

# Setup
```html
  <button 
    hx-get="/styles"
    hx-swap="outerHTML"
  ></button>
```

```typescript
// nestjs
@Get('/styles')
styles(@Res() res: Response) {
  res.setHeader('Content-Type', 'text/html');
  res.send('<style></style>');
}
```

# Expected
`<button>` to be replaced with `<style>`
```html
<style></style>
```

# Actual
```html
  <button 
    hx-get="/styles"
    hx-swap="outerHTML"
  ></button>
```

# Workaround
Wrap the `<style>` in any other tag  and the node will populate:
```typescript
// nestjs
@Get('/styles')
styles(@Res() res: Response) {
  res.setHeader('Content-Type', 'text/html');
  res.send('<span><style></style></span>');
}
```

Results in:
```html
<span><style></style></span>
```

**Comment by jsonberry** [comment]
Early investigation looks like this could be related to some checks on `NODE_TYPE`

Still very early on in the research, it's my first time checkin' out the source.

**Comment by najamkhn** [comment]
Just a guess here but changing the mime type to `text/css` and entirely skipping `<style />` in response be any help if its checking for `NODE_TYPE`?

**Comment by vdsabev** [comment]
This bug also occurs if `<style>` is the first in a list of sibling nodes like this:
```html
<style></style>
<div></div>
```
But not if it's second or third or last:
```html
<div></div>
<style></style>
```

**Comment by infogulch** [comment]
Does this behave differently if `htmx.config.useTemplateFragments` [config](https://htmx.org/docs/#config) is true?

**Comment by dmgd** [comment]
Maybe related? https://github.com/bigskysoftware/htmx/pull/1768

**Comment by dmgd** [comment]
Yeah, #1768 fixes this from what I can see (I just deleted the <html> stuff from the .html files, added a <br/> to one, like in https://github.com/bigskysoftware/htmx/issues/1771#issuecomment-1711140665, and then tried it out by clicking around)

**Comment by vtohr** [comment]
> Does this behave differently if `htmx.config.useTemplateFragments` [config](https://htmx.org/docs/#config) is true?

<html>
<body>


<script>
--
  | htmx.config.useTemplateFragments  = true;
  | </script>


</body>
</html>
fixed it just like you suggested.. thanks

**Comment by infogulch** [comment]
#1768 is merged, is this fixed?

**Comment by xhaggi** [comment]
I think this issue will also benefit from https://github.com/bigskysoftware/htmx/pull/1794

**Comment by alexpetros** [comment]
Resolved by #1768
